### PR TITLE
Respect bigquery_opts and update output path

### LINKED ISSUE #7: Overgenerates and doesn't respect paths=source_relative
Testing this next to `protoc-gen-bq-schema`, there were two initial problems that kept me from moving forward:
1. I expected schemas to be opt-in, and only generated when an option was set
`option (gen_bq_schema.bigquery_opts).table_name = "bar_table";`
2. Everything was generated to the same directory, not to the expected destination directories where all my other plugins generate code

`buf.yaml`

```yaml
  - name: bq-schema
    out: gen/go
    opt: paths=source_relative

  - name: bigquery # v0.1.2
    out: gen/go
    opt: paths=source_relative
```

### PR DESCRIPTION
**Pull request by mfridman**
Fix #7

This PR has 2 notable changes

## bigquery changes

1. Instead of outputting everything into the same directory, let's instead output the file into a directory structure that matches the package name. (This makes the plugin a drop-in replacement for `bq-schema`).

Note, this is NOT the equivalent of Go's `paths=source_relative` because we're building a path based on the package and not deriving it from the input path. This will be irrelevant for _most_ users because a `package foo.bar.baz.v1` will result in `foo/bar/baz/v1` (which is enforced with `buf`'s linter).

2. Buf default, only generate messages with bigquery_opts options. This is configurable internally, in the future we can surface this as a plugin option like `geenrate_all_messages`.

#### Before

```
./internal/testdata/bigquery
├── buf.protoschema.test.v1.BigQueryWellknownTypeTest.TableSchema.json
├── buf.protoschema.test.v1.BigQueryWellknownTypeTest.bigquery.proto
├── buf.protoschema.test.v1.CustomBigQuery.TableSchema.json
├── buf.protoschema.test.v1.CustomBigQuery.bigquery.proto
├── buf.protoschema.test.v1.CustomOptions.TableSchema.json
├── buf.protoschema.test.v1.CustomOptions.bigquery.proto
├── buf.protoschema.test.v1.NestedReference.TableSchema.json
├── buf.protoschema.test.v1.NestedReference.bigquery.proto
├── bufext.cel.expr.conformance.proto3.TestAllTypes.TableSchema.json
└── bufext.cel.expr.conformance.proto3.TestAllTypes.bigquery.proto
```

#### After

- Filename matches the table name from `gen_bq_schema.bigquery_opts` .
- Output matches the package name, basically `.` replaced with `/`

```proto
message CustomBigQuery {
  option (gen_bq_schema.bigquery_opts) = {
    table_name: "CustomName"
    use_json_names: true
  };

  int32 int32_field = 1 [(gen_bq_schema.bigquery) = {
    type_override: "TIMESTAMP",
    name: "create_time"
  }];
  string string_field = 2;

  NestedReference nested_reference = 3 [(gen_bq_schema.bigquery) = {ignore: true}];
}
```

```diff
./internal/testdata/bigquery
-├── buf.protoschema.test.v1.CustomOptions.TableSchema.json
└── buf
    └── protoschema
        └── test
            └── v1
+                └── CustomName.TableSchema.json
```

## testing

I removed the ./internal/cmd binaries that generated the golden test files in the testdata directory. I think now that we have these as plugins we should have one implementation for code generation. We can still use the generated ./internal/gen for unit tests.

proto -> input (code generation request) -> output .. compare output to the golden testdata.

### CODE REVIEW

File: internal/protoschema/plugin/pluginbigquery/pluginbigquery.go [hunk 1]
```diff
@@ -47,8 +56,10 @@ func Handle(
 			if len(data) == 0 || string(data) == "null" {
 				continue
 			}
+			name := tableName + "." + bigquery.FileExtension
```
Comment (mfridman) [inline internal/protoschema/plugin/pluginbigquery/pluginbigquery.go:59]: For this to truly be a drop-in replacement for `bq-schema` we'd need to make the suffix customizable or just use whatever they do. I.e., we have it as `*.TableSchema.json` whereas they use `*.schema`.

**Review by Alfus — APPROVED** [review]

File: internal/protoschema/plugin/pluginbigquery/pluginbigquery.go (hunk 1 above)
Comment (derekperkins) [reply internal/protoschema/plugin/pluginbigquery/pluginbigquery.go]: FWIW, I'd prefer a `.json` extension, even if just for file formatting. Calling it `.schema` doesn't add much clarity IMO, and was strange when I started using their generator. Not a big deal either way.

**Comment by bufdev** [comment]
This is using a custom option from a repository we don't control and isn't that well-formed - before we commit to this, is this worth a discussion? Perhaps it's the right answer, but I don't want to commit to this without being sure we want to - compatibility with another plugin isn't a guarantee we need to make.

File: internal/protoschema/plugin/pluginbigquery/pluginbigquery.go (hunk 1 above)
Comment (mfridman) [reply internal/protoschema/plugin/pluginbigquery/pluginbigquery.go]: Thanks for your feedback @derekperkins. 

We've been chatting internally about this plugin and whether we want to support the existing custom options from bq_schema, create our own or support both for compatibility.

Likewise, there's the decision of where to output the files, because we want users to have a consistent experience using the plugins in this repository.

TL;DR - the plugins are still fairly alpha and we plan to revisit some specific behavior in the next few weeks. But wanted to give you an update in case you were wondering what's up with this plugin.

Comment (derekperkins) [reply internal/protoschema/plugin/pluginbigquery/pluginbigquery.go]: No problem. `protoc-gen-bq-schema` is working just fine for me at the moment. I'm a big buf fan, so I was happy to test it out. My question in #6 was to understand if there was a specific end goal different for this plugin, or if it was just a cleaner, more reliable implementation. I'm more than happy for you to take your time architecting, and will give feedback along the way

**Comment by mfridman** [comment]
We deprecated the protoschema-bigquery plugin, and removed it in https://github.com/bufbuild/protoschema-plugins/pull/12.

Closing this PR as it's not longer relevant.
